### DISCUSSION TITLE
v1.4.1

### DISCUSSION THREAD

**Discussion by eduardofilo** [eduardofilo/RG350_adam_image#199] [Announcements]
## Change log

Minor version to fix some problems found in 1.4 version since publishing. The changes introduced are:

* Added `.zip` and `.bin` to Supervision supported extensions (issue #25).
* Added PocketSNES to Py Backup configuration (issue #179).
* Added config dir of FBA to Py Backup configuration thanks to @neilswann80.
* Added alias file to PICO-8 system thanks to @neilswann80.
* Modified the system that creates link files for Quake and Quake2 to be smarter thanks to @neilswann80.

Follow installation/upgrade instructions on 1.4 release description.

The `upgrade_1.4.1.tgz` file allows to update version 1.4 to 1.4.1 without having to flash again. To do this, proceed as follows:

1. Copy the `upgrade_1.4.1.tgz` file to the root of the external card.
2. Insert the external card in the console and boot. 
3. Connect console to PC with USB cable.
4. Open a terminal in Windows or Linux and initiate a SSH session with command `ssh od@10.1.1.2` (the password is in [README>System access](https://github.com/eduardofilo/RG350_adam_image/#system-access)).
5. Untar the file copied in step 1 with this command: `sudo tar -xzf /media/sdcard/upgrade_1.4.1.tgz -C /` (same password than in step 4).
6. You are done.

The `config.ini` file is related with issue #179. In v1.1 was installed the standalone PocketSNES emulator, but it was not incorporated into the Py Backup configuration by mistake, so if savestates have been generated and you want to back them up with Py Backup, you must load that file following the video in the [Upgrade section](https://github.com/eduardofilo/RG350_adam_image#upgrade) of the README.

<hr /><em>This discussion was created from the release <a href='https://github.com/eduardofilo/RG350_adam_image/releases/tag/v1.4.1'>v1.4.1</a>.</em>

**Comment by valzi** [comment]
Thank you!

**Comment by Chris3911** [comment]
Hello I would like to know in my case. I make a clean install of the 1.4.its really necessary to use the config.ini I don't really have any safes at all I'm starting from scratch

**Reply by neilswann80** [reply]
@Chris3911 if you install 1.4.1 as a fresh install (without using Py Backup) you won't need to worry about config.ini being overwritten by an older backup.  You say you don't have any saves to worry about, so the lack of SNES backups won't affect you either.  You're good to go!

@eduardofilo Thinking about it... Does Py Backup restore/overwrite entire config files, or merge them together, adding missing lines for example?

**Reply by Chris3911** [reply]
And if I want to install a newer version in the future do I have to worry about that or is going to be fine

**Reply by neilswann80** [reply]
In the future I think you'll only ever need to worry about this config.ini (Py Backup settings) if you use Py Backup to backup itself... or you've modified the config.ini file yourself to add backups for standalone apps and game ports (personally added, not part of the standard image).

For example I've personally added standalone games like DevilutionX, OpenLara, SuperMario64 to my system and have modified the config.ini to also automate the backup of my savegames for these additional games via Py Backup.

If you're only using the Py Backup to backup/restore the "Adam Image" stock apps/emulators you should be A-OK.

Hopefully that makes sense?  Apologies if not, I'm running on low sleep atm.

**Reply by Chris3911** [reply]
It's fine I understand that's good because I will stick to Adam image I like it a lot and that's the device I have the 280v which I love with all my heart and yes the performance is great with this Adam. What I like the most is that I believe correct me if I'm wrong this images in every new release comes with same apps or maybe some other but updated I believe Adam image always update their emulators inside the build right?

**Reply by neilswann80** [reply]
Yes, eduardofilo has done an amazing job putting this image together and also regularly updates the emulators and apps within it.  Occasionally he'll downgrade an app if the newer version is found not to work as well as an older version, this happened recently with DOSBox.

The 280v is a great choice, I may get one myself.  I'm rocking an RG350M, but an even more pocketable handset would be welcome.  This image really has given these handhelds a new lease of life, like having a brand new device again!

**Reply by Chris3911** [reply]
> Yes, eduardofilo has done an amazing job putting this image together and also regularly updates the emulators and apps within it.  Occasionally he'll downgrade an app if the newer version is found not to work as well as an older version, this happened recently with DOSBox.
> 
> The 280v is a great choice, I may get one myself.  I'm rocking an RG350M, but an even more pocketable handset would be welcome.  This image really has given these handhelds a new lease of life, like having a brand new device again!  

I totally agree 💯👍

**Reply by eduardofilo** [reply]
Hi @Chris3911. @neilswann80 explained very well, but I would like to make some clarifications.
The intention is that each new version of the image adapts the Py Backup configuration to suit the installed standalone emulators (RA does not change the directories where files worth backing up are stored, this is one of its many advantages).
But when PocketSNES was installed on v1.1, I forgot to add it. So if someone has used that emulator and saved games since that version of the image, they have no chance to backup the files if they don't manually install that file on the previous versions to migrate to 1.4.1.
But in general it shouldn't be necessary to manage that Py Backup configuration file manually unless you add emulators or ports as @neilswann80 says. That's why the config.ini file only appears in a couple of releases (in 1.2 it was also added, but for a different reason).

**Reply by Chris3911** [reply]
> Hi @Chris3911. @neilswann80 explained very well, but I would like to make some clarifications.
> The intention is that each new version of the image adapts the Py Backup configuration to suit the installed standalone emulators (RA does not change the directories where files worth backing up are stored, this is one of its many advantages).
> But when PocketSNES was installed on v1.1, I forgot to add it. So if someone has used that emulator and saved games since that version of the image, they have no chance to backup the files if they don't manually install that file on the previous versions to migrate to 1.4.1.
> But in general it shouldn't be necessary to manage that Py Backup configuration file manually unless you add emulators or ports as Neil says. That's why the config.ini file only appears in a couple of releases (in 1.2 it was also added, but for a different reason).

Thanks for take time and explain to us thank you again

**Comment by retrogamecorps** [comment]
Great job, it was a nice and simple upgrade process.  I'm going an RG280V vs Miyoo Mini comparison video soon, and so I'll be sure to mention the update in the video.

**Reply by neilswann80** [reply]
EDIT: The @retrogamecorps video mentioned: https://youtu.be/lq3mvzkNKZ0

Whilst you're both here...
I really appreciate the work you both ( @eduardofilo and @retrogamecorps  ) do for the gaming community.  It's been an "interesting" few years for everybody and having an avenue of escapism like gaming has been a welcome distraction; you've both had a very positive effect on my retro gaming enjoyment!
/end sycophantic blurb ;)

**Reply by Chris3911** [reply]
> Great job, it was a nice and simple upgrade process.  I'm going an RG280V vs Miyoo Mini comparison video soon, and so I'll be sure to mention the update in the video.

I will really like to see that video I was wondering which one is better

**Reply by neilswann80** [reply]
Same for me.  I have a feeling I'd prefer the feel/comfort of the MM over the 280V, particularly on longer gaming sessions (I have 3D printed grips for my 350 which are a godsend).  But I'm not sure if the seamless experience the Adam Image (with Simplemenu and Retroarch) can be beaten.  I look forward to Russ' comparison.

**Reply by Chris3911** [reply]
> Same for me.  I have a feeling I'd prefer the feel/comfort of the MM over the 280V, particularly on longer gaming sessions (I have a 3D printed grips for my 350 which are a godsend).  But I'm not sure if the seamless experience the Adam Image (with Simplemenu and Retroarch) can be beaten.  I look forward to Russ' comparison.

To be honest I prefer the look and design of the miyo.  but at the end both are awesome devices

**Reply by eduardofilo** [reply]
> Whilst you're both here... I really appreciate the work you both ( @eduardofilo and @retrogamecorps ) do for the gaming community. It's been an "interesting" few years for everybody and having an avenue of escapism like gaming has been a welcome distraction; you've both had a very positive effect on my retro gaming enjoyment! /end sycophantic blurb ;)

Well, I guess we can say you are part of the team.

Not forgetting all the developers of OpenDingux, RetroArch, SimpleMenu and the standalone emulators and utilities.

**Reply by eduardofilo** [reply]
Thanks Russ.
You make this look good.
"As always, thanks for reading..." 😉

**Reply by kjettern** [reply]
I now have both RG280V and Miyoo Mini. Personally the RG280V sits better in my hands than the Miyoo, but the better screen resolution and the look of Miyoo is very tempting. One drawback to the Miyoo is the lack of good software, so mr. Eduardofilo: Do you think it's possible with a Adam Image for the Miyoo Mini in the future? Thank you so much for the Adam Image! It's amazing!

**Reply by eduardofilo** [reply]
:-)
I am afraid it is not possible. That machine is ARM and therefore none of my experience can be used. And I have no plans to buy more machines or to continue working on these topics. I would like a change of scenery.

**Reply by kjettern** [reply]
I see! Sad that the Miyoo isn't based on the same cpu. I really hope you will continue to work on the Adam image a little bit more! Really love my Rg280v! You're doing a great job!

**Reply by valzi** [reply]
I hope someone will decide to take over when Eduardo is ready to step back.

**Reply by neilswann80** [reply]
@retrogamecorps https://youtu.be/lq3mvzkNKZ0

**Comment by eduardofilo** [comment]
Bug notification: For some reason that I still can't explain, the v1.4.1 image includes SimpleMenu version 10.2 and not 10.3 as it should (as for example v1.4 includes). The problems this causes are minor (the three new launchers added in v1.4 in the Apps section do not work). This weekend I will create a new release to fix this and include a patch to be applied on v1.4 or v1.4.1. The problem can also be fixed manually by copying the following file to the /media/data/apps directory. Sorry for the inconvenience.

https://github.com/fgl82/simplemenu/releases/download/10.3/SimpleMenu-OD-BETA.opk

**Reply by Chris3911** [reply]
Hola eso no es lo unico uno de los niveles de brillo ase parpadear la pantalla suave mente

**Reply by K9-711** [reply]
I installed the [SimpleMenu-OD-BETA.opk you linked and recognized that the size is exactly the same as in Adam 1.4.1.

**Reply by Chris3911** [reply]
Were im supposed to pui it on internal storage or sd card apps folder ?

**Reply by neilswann80** [reply]
Internal storage `/media/data/apps`

**Reply by eduardofilo** [reply]
> I installed the [SimpleMenu-OD-BETA.opk you linked and recognized that the size is exactly the same as in Adam 1.4.1.

The size is the same, right. But internally they are differents. I have checked it.

**Reply by jajabinx55** [reply]
Hi Eduardo,

First of all, it took me some time but I've finally found the place to thank you for yours and team hard work on Adam Image.

I just had a quick question regards to Retroarch Hatari core. I managed to get this to work and games are running very well. But i've noticed a few issues with the Hatari Menu (mainly, it doesn't 'save state' when selecting the 'save' option under 'memory').

Just out of curiousity, I was looking for the 'hatari_libretro.so' core file, to see if there's been any latest version released (to maybe see if the later version has fixed the version) but I can't seem to locate it anywhere....How did you manage to locate the Hatari Core (I couldn't see it on OpenDingux retroarch downloads either). Was really impressed to see that you managed to get it on the Adam build, but I want to see if I can troubleshoot by testing the latest version of the core and if it works, to report back (and maybe something you want to add to your future updates).

Many thanks in advance.

**Comment by eduardofilo** [comment]
@jajabinx55

> I just had a quick question regards to Retroarch Hatari core. I managed to get this to work and games are running very well. But i've noticed a few issues with the Hatari Menu (mainly, it doesn't 'save state' when selecting the 'save' option under 'memory').

I am afraid that core does not support savestates. This is what you see at the bottom of the screen when you try to generate one with the key combination Select+R1.
![screenshot004](https://user-images.githubusercontent.com/2923728/155881000-d1d5332a-025b-4fa8-8f87-1295c4abc1d4.png)

> Just out of curiousity, I was looking for the 'hatari_libretro.so' core file, to see if there's been any latest version released (to maybe see if the later version has fixed the version) but I can't seem to locate it anywhere....How did you manage to locate the Hatari Core (I couldn't see it on OpenDingux retroarch downloads either). Was really impressed to see that you managed to get it on the Adam build, but I want to see if I can troubleshoot by testing the latest version of the core and if it works, to report back (and maybe something you want to add to your future updates).

That core was one of the manual build that @Poligraf made several months ago:
https://github.com/Poligraf/opendingux_ra_cores_unofficial

Thanks for your words.
